libs/gltfio/src/ResourceLoader.cpp

```diff
@@ -985,46 +998,115 @@ void ResourceLoader::updateBoundingBoxes(FFilamentAsset* asset) const {
         tm.setParent(tm.getInstance(e), 0);
     }
 
-    auto computeBoundingBox = [](const cgltf_primitive* prim, Aabb* result) {
+    using SkinnedPrimitive = std::pair<cgltf_primitive*, const Skin*>;
+
+    const size_t posAttrSize = cgltf_num_components(cgltf_type_vec3);
+    const size_t skinningAttrSize = cgltf_num_components(cgltf_type_vec4);
+    const bool weightNormalized = pImpl->mNormalizeSkinningWeights;
+    const bool ignoreBindTransform = pImpl->mIgnoreBindTransform;
+    auto computeBoundingBox = [&](const SkinnedPrimitive skinnedPrimitive, Aabb* result) {
```

Comment:
This lambda is a bit difficult to read due to the the repeated `!ignoreBindTransform && skin` conditions and interleaved code.  I think this can be refactored into two separate functions: a `computeBoundingBox(const cgltf_primitive*, Aabb*)` which is identical to the current one, and a new `computeBoundingBoxSkinned(const cgltf_primitive*, const Skin*, Aabb*)` function that contains the new skinning logic.

Then, the logic below that invokes `computeBoundingBox` can decide which one to invoke based on if skinning is present and flags like `ignoreBindTransform`.